### PR TITLE
[Mailer] require a body renderer only if message has not been rendered yet

### LINKED ISSUE #48472: Update to 6.2 breaks email sending.
### Symfony version(s) affected

6.2

### Description

After updating to 6.2 I got this message:

`Handling "Symfony\Component\Mailer\Messenger\SendEmailMessage" failed: You must configure a "Symfony\Component\Mime\BodyRendererInterface" when a "Symfony\Bridge\Twig\Mime\NotificationEmail" instance has a text or HTML template set. (500 Internal Server Error)`

But when I try:

`$this->bodyRenderer->render($message);`

I get the error:

`TypeError: Symfony\Bridge\Twig\Mime\BodyRenderer::render(): Argument #1 ($message) must be of type Symfony\Component\Mime\Message, Symfony\Component\Notifier\Message\EmailMessage given`

### How to reproduce

NewNotificationByEmail.php

```
...
class NewNotificationByEmail implements NewNotificationInterface
{
    ...

    public function __invoke(array $pdfsName, int $entityId, string $entityToken, string $clientEmail, string $template) 
    {
        ...
        try {
            $clientNotification = new ClientNewNotificationByEmail($entityToken, $attachments, $template);
            $this->notifier->send($clientNotification, new Recipient($clientEmail));
        } catch (Exception $e) {
            $this->logger->critical('Failed to notify client', ['error'=>$e->getMessage(), 'code'=>$e->getCode(), 'file'=>$e->getFile(), 'line'=>$e->getLine(), 'id'=>$entityId]);
            throw $e;
        }
    }
}
```

ClientNewNotificationByEmail.php

```
...
class ClientNewNotificationByEmail extends Notification implements EmailNotificationInterface {
    ...

    public function asEmailMessage(EmailRecipientInterface $recipient, string $transport = null): ?EmailMessage
    {
        $message = EmailMessage::fromNotification($this, $recipient, $transport);

        $message->getMessage()
            ->htmlTemplate($this->template)
            ->context(['entityToken' => $this->entityToken, 'footer_text'=>null])
            ->importance('')
        ;

        foreach ($this->attachments as $att) {
            $message->getMessage()->attachFromPath($att['path'], $att['name'], 'application/pdf');
        }

        return $message;
    }
}
```

### Possible Solution

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 6.2
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #48472
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Symfony/Component/Mailer/Transport/AbstractTransport.php [hunk 1]
```diff
@@ -76,7 +76,7 @@ public function send(RawMessage $message, Envelope $envelope = null): ?SentMessa
             $envelope = $event->getEnvelope();
             $message = $event->getMessage();
 
-            if ($message instanceof TemplatedEmail && ($message->getTextTemplate() || $message->getHtmlTemplate())) {
+            if ($message instanceof TemplatedEmail && (!$message->getTextBody() && $message->getTextTemplate() || !$message->getHtmlBody() && $message->getHtmlTemplate())) {
```
Comment (fabpot) [inline src/Symfony/Component/Mailer/Transport/AbstractTransport.php:79]: Would this be enough?

```suggestion
            if ($message instanceof TemplatedEmail && (null !== $message->getTextTemplate() || null !== $message->getHtmlTemplate())) {
```

Comment (xabbuh) [reply src/Symfony/Component/Mailer/Transport/AbstractTransport.php:79]: I don't think so. Both methods are overwritten in the `NotificationEmail` class: https://github.com/symfony/symfony/blob/6.2/src/Symfony/Bridge/Twig/Mime/NotificationEmail.php#L158-L174

Comment (ThomasTr) [reply src/Symfony/Component/Mailer/Transport/AbstractTransport.php:79]: for me

```
(null !== $message->getTextTemplate() || null !== $message->getHtmlTemplate())
````
is `true`, `$message->getTextTemplate()` is `@email/default/notification/body.txt.twig` and `$message->getHtmlTemplate()` is `@email/default/notification/body.html.twig`

Comment (fabpot) [reply src/Symfony/Component/Mailer/Transport/AbstractTransport.php:79]: Let me have a look.

Comment (NCSantos) [reply src/Symfony/Component/Mailer/Transport/AbstractTransport.php:79]: For my issue [https://github.com/symfony/symfony/issues/48472](https://github.com/symfony/symfony/issues/48472) this fixes it:

`if ($message instanceof TemplatedEmail && (!$message->getTextBody() && $message->getTextTemplate() || !$message->getHtmlBody() && $message->getHtmlTemplate())) {`

**Comment by fabpot** [comment]
See #48505 for an alternative fix.

File: src/Symfony/Component/Mailer/Transport/AbstractTransport.php (hunk 1 above)
Comment (fabpot) [reply src/Symfony/Component/Mailer/Transport/AbstractTransport.php:79]: See #48505 for an alternative fix.

**Comment by xabbuh** [comment]
closing in favor of #48505

**Pull request closed without merging**